sugarcrm/customer.py

```diff
@@ -0,0 +1,29 @@
+import sugarcrm
+from sugarcrm.client import APIClient
+
+
+class Customer:
+    def __init__(self, logger=sugarcrm.logger):
+        self.logger = logger
+        self.roles = set()
+        self.api_client = APIClient()
+
+    def fetch(self, email):
+        # In SugarCRM personal individuals are in Accounts module and organizations in Contacts
+
+        account_id, contact_id = self.api_client.search_email(email)
+        roles_as_account = roles_as_contact = ""
+        if account_id:
+            roles_as_account = self.api_client.get_field(
+                "Accounts", account_id, "stic_relationship_type_c"
+            )
+        if contact_id:
+            roles_as_contact = self.api_client.get_field(
+                "Contacts", contact_id, "stic_relationship_type_c"
+            )
+        self.roles = set(roles_as_account.split(",") + roles_as_contact.split(","))
+        if "" in self.roles:
+            self.roles.remove("")
+        self.logger.debug("Customer {} has the following roles {}".format(email, self.roles))
```

Comment:
This might be useful while developing but I'm not sure it will add any value in production logs. What's more, I think it'll add noise and make logs unnecessary large. We mustn't forget we have a daily quota and we'll have to pay more if we exceed it.

Reply:
One doubt, if our default production logging mode is INFO in settings.py , the debug emails won't be stored in the online logs isn't ?

Reply:
If I understood correctly, you mean that particular email that line 27 is logging, right? in that case, no, right.